sample/rules.go

```diff
@@ -158,10 +173,18 @@ func ruleMatchesSpanInTrace(trace *types.Trace, rule *config.RulesBasedSamplerRu
 		for _, condition := range rule.Condition {
 			// whether this condition is matched by this span.
 			value, exists := extractValueFromSpan(span, condition, checkNestedFields)
+			if condition.Matches == nil {
+				if !conditionMatchesValue(condition, value, exists) {
+					ruleMatched = false
+					break // if any condition fails, we can't possibly succeed, so exit inner loop early
+				}
+			}
 
-			if !conditionMatchesValue(condition, value, exists) {
-				ruleMatched = false
-				break // if any condition fails, we can't possibly succeed, so exit inner loop early
+			if condition.Matches != nil {
```

Comment:
If we make it here we know `condition.Matches != nil` so I believe we replace this if statement with an else if statement.

Reply:
sweet. updated.